src/components/map/MapDirective.js

```diff
@@ -80,6 +81,40 @@ goog.require('ga_permalink');
 
             map.setTarget(element[0]);
 
+            scope.$watch('::ol3d', function(ol3d) {
+              if (ol3d) {
+                var camera = ol3d.getCesiumScene().camera;
+                var params = gaPermalink.getParams();
+
+                // initial location based on query params
+                var position = undefined;
+                if (params.lon && params.lat && params.elevation) {
+                  var lon = parseFloat(params.lon);
+                  var lat = parseFloat(params.lat);
+                  var elevation = parseFloat(params.elevation);
+                  position = new Cesium.Cartographic(lon, lat, elevation);
+                }
+                camera.setView({
+                  positionCartographic: position,
+                  heading: params.heading ? parseFloat(params.heading) : undefined,
+                  pitch: params.pitch ? parseFloat(params.pitch) : undefined,
+                  roll: 0.0
+                });
+
+                // update permalink
+                camera.moveEnd.addEventListener(function() {
```

Comment:
This is a good point. @fredj, have you verified how often the listener is called? `gaDebounce` is already used for the update of the 2D map parameters, so it probably makes sense to do the same for the 3D params.